### PR TITLE
feat(core): Add hook with logic to check if an asset is 'Transferable'.

### LINKED ISSUE #3962: Update `isAssetTransferable` logic to use `getNormalizedMoveStruct`
PR https://github.com/iotaledger/iota/pull/3831 removed `hasPublicTransfer`, we should update the logic used by the tooling apps so that the `isAssetTransferable` logic is correct again

We should now use https://docs.iota.org/references/ts-sdk/api/client/classes/IotaClient#getnormalizedmovestruct per object, and if the `Store` **ability** is included, then the object can be transferred

### PR DESCRIPTION
**Pull request by msarcev**
# Description of change

- Add new hook way
- Remove old util function way

## Links to any relevant issues

Closes https://github.com/iotaledger/iota/issues/3962

### CODE REVIEW

**Review by brancoder — DISMISSED** [review]
conflicts :pray:

**Review by begonaalvarezd — APPROVED** [review]
nice

**Review by panteleymonchuk — APPROVED** [review]

File: apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx [hunk 1]
```diff
@@ -8,21 +8,23 @@ import { useOwnedNFT } from '_hooks';
...
     const { data: ownedNFT, isPending: isNftLoading } = useOwnedNFT(nftId || '', address);
+    const { data: isAssetTransferable, isLoading: isCheckingAssetTransferability } =
+        useIsAssetTransferable(ownedNFT);
     const navigate = useNavigate();
     const isGuardLoading = useUnlockedGuard();
     const isPending = isNftLoading || isGuardLoading;
     return (
         <Overlay showModal title="Send NFT" closeOverlay={() => navigate('/nfts')} showBackButton>
-            <Loading loading={isPending}>
+            <Loading loading={isPending || isCheckingAssetTransferability}>
```
Comment (marc2332) [inline apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx]:
```suggestion
            <Loading loading={isPending}>
```

File: apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx [hunk 2]
```diff
@@ -8,21 +8,23 @@ import { useOwnedNFT } from '_hooks';
...
 function NftTransferPage() {
     const { nftId } = useParams();
     const address = useActiveAddress();
     // verify that the nft is owned by the user and is transferable
     const { data: ownedNFT, isPending: isNftLoading } = useOwnedNFT(nftId || '', address);
+    const { data: isAssetTransferable, isLoading: isCheckingAssetTransferability } =
+        useIsAssetTransferable(ownedNFT);
     const navigate = useNavigate();
     const isGuardLoading = useUnlockedGuard();
     const isPending = isNftLoading || isGuardLoading;
```
Comment (marc2332) [inline apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx]:
```suggestion
    const isPending = isNftLoading || isGuardLoading || isCheckingAssetTransferability;
```

File: apps/core/src/hooks/useIsAssetTransferable.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            const structAbilities = moveNormalizedStruct?.abilities?.abilities ?? null;
+
+            if (!structAbilities) {
+                return false;
+            }
+
+            return structAbilities.includes('Store');
+        },
+    });
```
Comment (marc2332) [inline apps/core/src/hooks/useIsAssetTransferable.ts:49]: Not that it makes a huge difference but I think it is a good practice.
```suggestion
		enabled: !!(packageId && moduleName && functionName),
    });
```

**Review by marc2332 — DISMISSED** [review]

File: apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx (hunk 1 above)
Comment (msarcev) [reply apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx]: Why ?

File: apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx (hunk 2 above)
Comment (msarcev) [reply apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx]: Ah, i see why :)

**Pull request merged**